### ISSUE TITLE
No audio notification in Arch Linux

### ISSUE THREAD

**Issue by yochaigal** [wavebox/waveboxapp#340]
* Which version of Wavebox are you using? [e.g. 3.0.0]
3.2.0

* Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]
Arch Linux 64bit with KDE Plasma 5.10

Everything works great, except that notification audio doesn't work (Default or Enhanced).

**Comment by rhys101** [comment]
Which method did you use to install Wavebox?

Just tested here on an Arch Linux 64 bit (2017.08.01) Plasma 5.10 with Wavebox from tar - selecting `Enhanced`, `Play notification sound` and a sound of `Beep`. 

Do you hear anything when clicking the `Test Notification` in settings?

**Comment by yochaigal** [comment]
I tested it from Tar, and from AUR. And no, it doesn't make a sound when I
test notifications either.

I'll try it on another Arch machine I have as well.

**Comment by yochaigal** [comment]
So, on my X1 Carbon (also running Arch with KDE) the default notification sound doesn't work, but the Beta (experimental) one does! I really like that one, actually - too bad it doesn't work on my desktop. The only real difference between the two is that my desktop has an nVidia GPU. 

In a strange (possibly relevant) twist, on my laptop Wavebox works great with only libappindicator-gtk2 and libappindicator-gtk3; the icon looks good AND it raises on left click. On my work machine left-click does not work. Just an aside.

**Comment by Thomas101** [comment]
@yochaigal sound like libappindicator on the Default provider. When using the enhanced notification handler it uses html5 audio to play the sound. Would you be able to upload a m4a file to google drive and see if it plays?

**Comment by yochaigal** [comment]
mp3 files play fine as gmail attachments; I can play the m4a as a preview in Google Drive (within wavebox) .

**Comment by yochaigal** [comment]
Sorry this took so long!
With every version tested (including beta 3.3.2) I have no sound on my work PC. I cannot play attachments, nor does the default/experimental notifications create sound (the latter doesn't create a notification at all).

**Comment by Thomas101** [comment]
It sounds like the audio library that electron uses isn't installed. How did you install Wavebox, was it by deb, tar, snap, other? If you try playing audio from your browser does that work & which browser is it?

**Comment by Thomas101** [comment]
e.g. if you try playing [the audio file here](https://www.w3schools.com/html/tryit.asp?filename=tryhtml5_audio_all) from chrome

**Comment by yochaigal** [comment]
I installed it through the Arch User Repository (AUR), exactly the way I
installed it on my other two machines.
That audio file plays just fine in Chrome.

To be clear, I have Arch Linux on all three of my machines (work machine,
home laptop, home desktop). This only happens with my work machine.

**Comment by yochaigal** [comment]
I'm unable to use this product at work, unfortunately - I have to listen to
voicemails on the web, and not being able to do so is sort of forcing my
hand. Just FYI (I'm a subscriber).

**Comment by Thomas101** [comment]
Seeing that it's working on two installs but not the other, it sounds like it could be a config issue on the machine. It's probably worth checking the following...

1. Open `System Settings` then `Multimedia` then click on the `Audio Volume` section on the left.
2. Does the dialog show the ports drop-down in the `Outputs` section? (For example in the screenshot below I have multiple input ports, but only one output port).
  ![screen shot 2017-09-18 at 15 50 57](https://user-images.githubusercontent.com/103586/30548298-b0728e4e-9c89-11e7-8a66-1729e5b26f5c.png)
3. If you do have multiple output ports, can you switch between them and see if any are muted or have the volume all the way down? If so ensure they're all cranked up.
4. Next switch over to the `Applications` tab and leave this window open.
5. Back in Wavebox open the Settings and ensure that under the `Notification Provider` section `Enhanced` is selected and pick one of the Notification Sounds which is quite obtrusive. `Correct` is a good example. Your settings should look something like this...
  ![screen shot 2017-09-18 at 15 57 01](https://user-images.githubusercontent.com/103586/30548450-17d6e8c8-9c8a-11e7-80ee-10540fc558ef.png)
6. Next, press the `Test Notification` button. In the System settings do you now see Wavebox listed under the `Applications` tab? What's the volume setting been configured to for that?
  ![screen shot 2017-09-18 at 15 53 13](https://user-images.githubusercontent.com/103586/30548503-3c7ff98a-9c8a-11e7-9254-5cac1ccbad62.png)
7. Next, try clicking on the `Audio and Video` section on the left in the System settings. Under the `Audio Playback` tree, are all sections listed to playback on the `Default` audio device or something different?
  ![screen shot 2017-09-18 at 16 02 13](https://user-images.githubusercontent.com/103586/30548702-cf3745da-9c8a-11e7-9095-5a0eba6332c7.png)

The other thing to do here, is to see if we can get any more info from `pacmd`. If you fire up a terminal and type `pacmd list-sink-inputs` with no audio playing you should get `0 sink input(s) available`. If you hit the `Test Notification` button as in step 6 and quickly run `pacmd list-sink-inputs` whilst it's still playing you should get a log similar to the following....
![screen shot 2017-09-18 at 16 17 40](https://user-images.githubusercontent.com/103586/30549550-f91ec1fa-9c8c-11e7-83e0-bc57e6931f2b.png)

what's the output of that? If the log shows Wavebox as being muted or having a different volume sound, can you make a note of the index of Wavebox (in the screenshot below it was 3) and run the following in terminal (replacing <index> with your index)

`pacmd set-sink-input-mute <index> false`
`pacmd set-sink-input-volume <index> 0x10000`

and try testing the notification again do you hear any audio?

**Comment by yochaigal** [comment]
First, thank you SO much for the in-depth response!

I was able to get audio _within_ the application to work by switching inputs in pavucontrol (thanks, I can't believe I didn't check) but notifications still do not work! When I press "test" while running pacmd list-sink-inputs I actually get 2 applications listed no matter what (speech-dispatcher). However, pressing Test Notification and running the same command shows no new outputs!

Then I did another test: I asked someone to email me, and it now DOES make a sound! I think the "test" function is just broken for some reason? I also don't get an OSD notification when pressing test, if that helps. 

Anyways my issues are cleared up - I can listen to voicemails in gmail/drive (my bad) and sounds do work  when emails/slack messages come in  (probably because I fixed it in pulseaudio).

Thanks a lot!
